### PR TITLE
ASM-8648: Added CD/DVD drive while creating vm

### PR DESCRIPTION
**Pull request by arunrordell**
Previously ASM did not have ability to vreate virtualCD drive to
support os_install via iso image.

This PR will include CD drive while creating vm.

### CODE REVIEW

File: lib/puppet/provider/vc_vm/default.rb [hunk 1]
```diff
@@ -619,6 +620,33 @@ def disk_backing(file_name="")
...
+        :backing => RbVmomi::VIM.VirtualCdromRemoteAtapiBackingInfo(:deviceName => "CDROM"),
+        :connectable => virtualcd_connect_info,
+        :controllerKey => 200,
+        :key => 3000,
+        :unitNumber => 0
+    )
+
+    config = {
+        :device => disk,
+        :operation => RbVmomi::VIM.VirtualDeviceConfigSpecOperation('add')
```
Comment (kaushalgala) [inline lib/puppet/provider/vc_vm/default.rb]: double quotes

File: lib/puppet/provider/vc_vm/default.rb [hunk 2]
```diff
@@ -619,6 +620,33 @@ def disk_backing(file_name="")
     )
   end
 
+  # spec for creating virtualCdRom
+  # it creates empty CDROM each time creating new_vm
+  def disk_spec_cd
```
Comment (kaushalgala) [inline lib/puppet/provider/vc_vm/default.rb]: better to call it cd_spec or cdrom_spec

File: lib/puppet/provider/vc_vm/default.rb [hunk 3]
```diff
@@ -619,6 +620,33 @@ def disk_backing(file_name="")
...
+    disk = RbVmomi::VIM.VirtualCdrom(
+        :backing => RbVmomi::VIM.VirtualCdromRemoteAtapiBackingInfo(:deviceName => "CDROM"),
+        :connectable => virtualcd_connect_info,
+        :controllerKey => 200,
```
Comment (kaushalgala) [inline lib/puppet/provider/vc_vm/default.rb]: lets add comment for why we chose 200 as controller key i.e. # IDE Controllers start at 200

**Comment by arunrordell** [comment]
Reference PR 
https://github.com/dell-asm/razor-server/pull/132

**Comment by gavin-scott** [comment]
:+1: Are you good @kaushalgala ?

**Pull request merged**